crates/uv/src/commands/python/list.rs

```diff
@@ -110,8 +135,14 @@ pub(crate) async fn list(
             .transpose()?
             .into_iter()
             .flatten()
-            // TODO(zanieb): Add a way to show debug downloads, we just hide them for now
-            .filter(|download| !download.key().variant().is_debug());
+            .filter(|download| {
+                all_variants
+                    || request
+                        .as_ref()
+                        .and_then(uv_python::PythonRequest::variant)
+                        .unwrap_or(uv_python::PythonVariant::Default)
+                        == *download.key().variant()
```

Comment:
I'm confused that we're changing the filtering here _and_ at https://github.com/astral-sh/uv/pull/16002/files#r2411618419

Why do we need both?